sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getApplicationProgressDetails.e2e-spec.ts

```diff
@@ -0,0 +1,135 @@
+  createE2EDataSources,
+  E2EDataSources,
+} from "@sims/test-utils";
+import {
+  ApplicationOfferingChangeRequestStatus,
+  ApplicationStatus,
+  Student,
+  StudentAppeal,
+  StudentAppealStatus,
+} from "@sims/sims-db";
+
+describe("ApplicationStudentsController(e2e)-getApplicationProgressDetails", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let studentAppealRepo: Repository<StudentAppeal>;
+  let student: Student;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    db = createE2EDataSources(dataSource);
+    studentAppealRepo = dataSource.getRepository(StudentAppeal);
+    student = await getStudentByFakeStudentUserType(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+      dataSource,
+    );
+  });
+
+  it("Should get the status of all requests and confirmations in student application (Appeals, Application Offering Change Requests and COE).", async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(
+      appDataSource,
+      { student },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+        createSecondDisbursement: true,
+      },
+    );
+    const [firstDisbursement, secondDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    // Create approved student appeal submitted yesterday.
+    const approvedAppealRequest = createFakeStudentAppealRequest();
+    const approvedAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [approvedAppealRequest],
+    });
+    await studentAppealRepo.save(approvedAppeal);
+    // Create pending student appeal submitted today.
+    const pendingAppealRequest = createFakeStudentAppealRequest();
+    pendingAppealRequest.appealStatus = StudentAppealStatus.Pending;
+    const pendingAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [pendingAppealRequest],
+    });
+    await studentAppealRepo.save(pendingAppeal);
+    // Create declined student appeal submitted today.
+    const declinedAppealRequest = createFakeStudentAppealRequest();
+    declinedAppealRequest.appealStatus = StudentAppealStatus.Declined;
+    const declinedAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [declinedAppealRequest],
+    });
+    await studentAppealRepo.save(declinedAppeal);
+    // Create approved application offering change request.
+    await saveFakeApplicationOfferingRequestChange(
+      db,
+      {
+        application,
+      },
+      {
+        initialValues: {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.Approved,
+        },
+      },
+    );
+    // Create pending application offering change request.
+    await saveFakeApplicationOfferingRequestChange(db, { application });
```

Comment:
mm, technically, we can't create an application offering request, when there is a pending request. and when we get the latest request status when it should be `pending` right (i.e. line 128)?